src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -194,13 +194,7 @@ object OrderedRDD {
     import kOk._
 
     import Ordering.Implicits._
-
-    rdd.partitioner match {
```

Comment:
OK, I think I know what to do here. Leave in this assert. and remove the `preservesPartitioning = true` from the other place you changed.